File: code-coaches/medium/splitting-strings/splitting_strings.py
def split_string(word: str, num_parts: int, separator: str = "-") -> str:
    parts = [word[i : i + num_parts] for i in range(0, len(word), num_parts)]
    return separator.join(parts)


def main():
    word = input()
    num_parts = int(input())
    print(split_string(word, num_parts))


if __name__ == "__main__":
    main()

File: code-coaches/medium/splitting-strings/SplittingStrings.cs
using System;

public class SplittingStrings
{
    static string SplitString(string word, int numParts, string separator = "-")
    {
        string[] parts = new string[(int)Math.Ceiling((double)word.Length / numParts)];
        for (int i = 0; i < parts.Length; i++)
        {
            int startIndex = i * numParts;
            int length = Math.Min(numParts, word.Length - startIndex);
            parts[i] = word.Substring(startIndex, length);
        }
        return string.Join(separator, parts);
    }

    static void Main(string[] args)
    {
        string word = Console.ReadLine();
        int numParts = int.Parse(Console.ReadLine());
        Console.WriteLine(SplitString(word, numParts));
    }
}
